platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml

```diff
@@ -39,12 +33,184 @@
   template:
     src={{ playbook_dir }}/../../../platforms/shared/configuration/roles/setup/edge-stack/templates/aes-custom-values.tpl
     dest={{ playbook_dir }}/../../../platforms/shared/configuration/build/aes-custom-values.yaml
+  when:
+    - network.type != 'indy'
+
+- name: Check if ambassador installed
+  k8s_info:
+    api_version: v1
+    kind: Service
+    name: ambassador
+    kubeconfig: "{{ kubeconfig_path }}"
+    namespace: "{{ proxy_namespace }}"
+  register: result
+
+- name: Set ambassador installed
+  set_fact:
+    ambassador_installed: "{{ result.resources|length > 0 }}"
+
+- name: Get available ports of installed ambassador
+  set_fact:
+    ports: "{{ result | json_query('resources[0].spec.ports[*].port') }}"
+  when: ambassador_installed
+
+- name: Format ambassador ports
+  args:
+    executable: /bin/bash
+  shell: |
+    json='{{ stewards | to_json }}'
+    length=$(echo "${json}" | jq '.[] | length')
+    index=0
+    declare -A ports
+    while [[ ${index} < ${length} ]]
+    do
+      steward=$( echo ${json} | jq ".[${index}]")
+      name=$(echo ${steward} | jq '.name' | tr -d '"')
+      node_port=$(echo ${steward} | jq '.node.ambassador' | tr -d '"')
+      client_port=$(echo ${steward} | jq '.client.ambassador' | tr -d '"')
+      if [[ ${name} != null ]]
+      then
+        if [[ ${ports["{{ kubecontext }}"]} != "" ]]
+        then
+          ports+=( ["{{ kubecontext }}"]+=, )
+        fi
+        ports+=( ["{{ kubecontext }}"]+=${node_port},${client_port} )
+      fi
+      index=$(( ${index} + 1 ))
+    done
+    echo ${ports["{{ kubecontext }}"]}
+  register: terminal
+  when:
+    - network['type'] == 'indy' and item.services.stewards is defined
+
+- name: Get Elastic IP
+  environment:
+    AWS_ACCESS_KEY_ID: "{{ aws.access_key }}"
+    AWS_SECRET_ACCESS_KEY: "{{ aws.secret_key }}"
+  shell: |
+    # format ip addresses list to string with space separator
+    ips=$(echo '{{ item.publicIps }}' | tr -d '["]' | sed 's/,/\ /g')
+    data=$(aws ec2 describe-addresses --public-ips ${ips} --region {{ aws.region }} --output json | jq '.Addresses[].AllocationId')
+    # format eip addresses list to string with comma separator (comma has to be escaped)
+    echo ${data} | tr -d '"' | sed 's/\ /\,/g'
+  register: allocation_ips
+  when:
+     - ( network.type == 'indy' and item.services.stewards is defined )
+     - item.cloud_provider == 'aws' or item.cloud_provider == 'aws-baremetal' #As this shell code is specific to AWS, cloud provider check has been added
+  tags:
+    - notest
+
+- name: Set Helm value for ambassador_eip
+  set_fact:
+    allocation_ips_stdout: "{{ allocation_ips.stdout | default('No value', true) }}"
+  when:
+    - allocation_ips is defined
+
+- name: Extract port numbers from the 'ports' string
+  set_fact:
```

Comment:
you are setting fact and registering it as well why ?

Reply:
where is the set fact extracted_ports being used ?

Reply:
updated the setfacts to use in node tpl's